super_calendar/models/super_calendar.py

```diff
@@ -0,0 +1,244 @@
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see http://www.gnu.org/licenses
+#
+##############################################################################
+
+from openerp import models, fields, api
+from openerp.tools.translate import _
+from mako.template import Template
+from datetime import datetime
+from openerp import tools
+from openerp.tools.safe_eval import safe_eval
+import logging
+
+
+def _models_get(self):
+    model_obj = self.env['ir.model']
+    model_ids = model_obj.search([])
+    return [(model.model, model.name) for model in model_ids]
+
+
+class SuperCalendarConfigurator(models.Model):
+    _logger = logging.getLogger(__name__)
+    _name = 'super.calendar.configurator'
+
+    name = fields.Char(
+        string='Name',
+        size=64,
+        required=True,
+    )
+    line_ids = fields.One2many(
+        comodel_name='super.calendar.configurator.line',
+        inverse_name='configurator_id',
+        string='Lines',
+    )
+
+    def _clear_super_calendar_records(self):
+        """ Remove old super_calendar records """
+        super_calendar_pool = self.env['super.calendar']
+        super_calendar_ids = super_calendar_pool.search([])
+        super_calendar_ids.unlink()
+
+    @api.multi
+    def generate_calendar_records(self):
+        """ At every CRON execution, every 'super calendar' data is deleted
+        and regenerated again. """
+
+        # Remove old records
+        self._clear_super_calendar_records()
+
+        # Rebuild all calendar records
+        configurator_ids = self.search([])
+        for configurator in configurator_ids:
+            for line in configurator.line_ids:
+                self._generate_record_from_line(configurator, line)
+        self._logger.info('Calendar generated')
+        return True
+
+    @api.multi
+    def _generate_record_from_line(self, configurator, line):
+        super_calendar_pool = self.env['super.calendar']
+        values = self._get_record_values_from_line(configurator, line)
+        for record in values:
+            super_calendar_pool.create(values[record])
+
+    @api.multi
+    def _get_record_values_from_line(self, configurator, line):
+        res = {}
+        current_pool = self.env[line.name.model]
+        domain = line.domain and safe_eval(line.domain) or []
+        current_record_ids = current_pool.search(domain)
+        for cur_rec in current_record_ids:
+            f_user = line.user_field_id and line.user_field_id.name
+            f_descr = (line.description_field_id and
+                       line.description_field_id.name)
```

Comment:
You don't need the `and` if `line.description_field_id` is an empty record set, `.name` will return false